### PR TITLE
Updated basic debugging

### LINKED ISSUE #14487: How do I print the model summary if .fit() is not called
## 📚 Documentation

QUESTION

[Sam](https://app.slack.com/team/U01GUP9HAGL)

Is there a way to print the model summary if .fit() is not called? I’m only calling .predict() but it’d still be useful to see the summary

______________________________________________________________________

#### If you enjoy Lightning, check out our other projects! ⚡

- [**Metrics**](https://github.com/Lightning-AI/metrics): Machine learning metrics for distributed, scalable PyTorch applications.

- [**Lite**](https://pytorch-lightning.readthedocs.io/en/latest/starter/lightning_lite.html): enables pure PyTorch users to scale their existing code on any kind of device while retaining full control over their own loops and optimization logic.

- [**Flash**](https://github.com/Lightning-AI/lightning-flash): The fastest way to get a Lightning baseline! A collection of tasks for fast prototyping, baselining, fine-tuning, and solving problems with deep learning.

- [**Bolts**](https://github.com/Lightning-AI/lightning-bolts): Pretrained SOTA Deep Learning models, callbacks, and more for research and production with PyTorch Lightning and PyTorch.

- [**Lightning Transformers**](https://github.com/Lightning-AI/lightning-transformers): Flexible interface for high-performance research using SOTA Transformers leveraging PyTorch Lightning, Transformers, and Hydra.

cc @Felonious-Spellfire @akihironitta

### PR DESCRIPTION
**Pull request by Felonious-Spellfire**
Updated basic debugging topic

## What does this PR do?

Adds missing content about printing the model summary if `.fit()` is not called. 

Fixes #14487

### Does your PR introduce any breaking changes? If yes, please list them.

None

## Before submitting

- [ ] Was this **discussed/approved** via a GitHub issue? (not for typos and docs)
- [ ] Did you read the [contributor guideline](https://github.com/Lightning-AI/lightning/blob/master/.github/CONTRIBUTING.md), **Pull Request** section?
- [ ] Did you make sure your **PR does only one thing**, instead of bundling different changes together?
- [X] Did you make sure to **update the documentation** with your changes? (if necessary)
- [ ] Did you write any **new necessary tests**? (not for typos and docs)
- [ ] Did you verify new and **existing tests pass** locally with your changes?
- [ ] Did you list all the **breaking changes** introduced by this pull request?
- [ ] Did you **update the CHANGELOG**? (not for typos, docs, test updates, or minor internal changes/refactors)

## PR review

Anyone in the community is welcome to review the PR.
Before you start reviewing, make sure you have read the [review guidelines](https://github.com/Lightning-AI/lightning/wiki/Review-guidelines). In short, see the following bullet-list:

- [X] Is this pull request ready for review? (if not, please submit in draft mode)
- [ ] Check that all items from **Before submitting** are resolved
- [ ] Make sure the title is self-explanatory and the description concisely explains the PR
- [ ] Add labels and milestones (and optionally projects) to the PR so it can be classified

## Did you have fun?

Make sure you had fun coding 🙃

cc @borda @Felonious-Spellfire @akihironitta

### CODE REVIEW

**Review by Borda — APPROVED** [review]

File: docs/source-pytorch/debug/debugging_basic.rst [hunk 1]
```diff
@@ -128,6 +128,15 @@ To add the child modules to the summary add a :class:`~pytorch_lightning.callbac
...
     trainer = Trainer(callbacks=[ModelSummary(max_depth=-1)])
 
+To print the model summary if ``.fit()`` is not called:
+
+.. testcode::
+
+    from pytorch_lightning.utilities import ModelSummary
+
+    summary = ModelSummary(model, max_depth=-1)
+    print(summary)
```
Comment (rohitgr7) [inline docs/source-pytorch/debug/debugging_basic.rst:138]:
```suggestion
    from pytorch_lightning.utilities.model_summary import ModelSummary

    model = LitModel()
    summary = ModelSummary(model, max_depth=-1)
    print(summary)
```

**Review by rohitgr7 — APPROVED** [review]

**Comment by rohitgr7** [comment]
@Felonious-Spellfire you can just write
```
Fixes #<issue_num_1>
Fixes #<issue_num_2>
```
in the PR description. This will link the issue to the PR, which will close automatically when it gets merged.

**Review by awaelchli — APPROVED** [review]

File: docs/source-pytorch/debug/debugging_basic.rst [hunk 2]
```diff
@@ -128,6 +128,15 @@ To add the child modules to the summary add a :class:`~pytorch_lightning.callbac
 
     trainer = Trainer(callbacks=[ModelSummary(max_depth=-1)])
 
+To print the model summary if ``.fit()`` is not called:
+
+.. testcode::
```
Comment (awaelchli) [inline docs/source-pytorch/debug/debugging_basic.rst:133]: This won't work, because model is not defined. Let's just use the regular code-block

```suggestion
.. code-block:: python
```

**Review by awaelchli — APPROVED** [review]
